### ISSUE TITLE
anomolous-backslash is noisy

### ISSUE THREAD

**Issue by nickdrozd** [PyCQA/pylint#5104]
### Bug description

```python
import re

print(re.sub('\.\.\.', '___', '_..._'))
```

Pylint flags each backslash as anomalous, even though they all appear in the same string. Add the `r` prefix to that one string fixes all three warnings.

(This isn't a correctness bug, but it is annoying.)

### Command used

```shell
pylint
```

### Pylint output

```
asdf.py:3:14: W1401: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
asdf.py:3:16: W1401: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
asdf.py:3:18: W1401: Anomalous backslash in string: '\.'. String constant might be missing an r prefix. (anomalous-backslash-in-string)
```

### Expected behavior

Issue just one warning for each string that needs the `r` prefix.

### Pylint version

```shell
pylint --version
pylint 2.11.2-dev0
astroid 2.8.0
Python 3.8.10 (default, Jun  2 2021, 10:49:15) 
```

**Comment by alanverdugo** [comment]
Hi, I took a look at the code, and this seems to be fixed by just `break`ing after the first "anomalous-backslash-in-string" message is added at https://github.com/PyCQA/pylint/blob/67957e233cdd402cacc4389aeba5e717a684f7a2/pylint/checkers/strings.py

Basically, adding a break at line 905, like this:
```
...
            elif next_char not in self.ESCAPE_CHARACTERS:
                self.add_message(
                    "anomalous-backslash-in-string",
                    line=line,
                    args=(match,),
                    col_offset=col_offset,
                )
                break
            # Whether it was a valid escape or not, backslash followed by
            # another character can always be consumed whole: the second
            # character can never be the start of a new backslash escape.
            index += 2
```

I did this and now the output is indeed just 1 warning instead of 3.

However, when I run the tests, a functional test fails.

```
    def runTest(self) -> None:
>       self._runTest()
E       AssertionError: Wrong results for file "excess_escapes":
E       
E       Expected in testdata:
E          8: anomalous-backslash-in-string
E         17: anomalous-backslash-in-string
E         19: anomalous-backslash-in-string

pylint/testutils/lint_module_test.py:87: AssertionError

```

I am not sure how to fix that failing test, or if this is an acceptable solution for this issue.

Sorry, this is my first time trying to help with pylint.

**Comment by DanielNoord** [comment]
I took a look at the the failing functional test and this is because the current test actually expects the message to be fired twice. You would thus need to change the test to also expect one message. This can be done in `tests/functional/e/excess_escapes.py`.

However, personally I would like this to be a toggle able option. The messages are correct and I would expect the default behaviour to be to report all instances of code that trigger a warning. Basically, I think the noisiness should be default behaviour and could suppressed by a toggle named `suppress-double-messages`.

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord is right about the way to fix the functional test.

I also agree with @DanielNoord here, but at the same time adding an option for this seem like a lot of work and potential problems for something very tiny and inconsequential. I think for this one we should, as maintainers, take a decision and stick with it. 

Let's vote:
Error for each anomalous backslash: 🚀 
Error for each string containing anomalous backslash: 🎉

**Comment by Pierre-Sassoulas** [comment]
Seeing the result of the vote after two weeks I'm going to close as "won't do"
